File: 프로그래머스/2/131704. 택배상자/택배상자.py
def solution(order):
    cnt,stack,cursor = 0,[],0
    
    for i in range(1,len(order)+1):
        stack.append(i)
        while stack and stack[-1] == order[cursor] :
            stack.pop()
            cursor+=1
            cnt+=1
       

    return cnt

File: 프로그래머스/2/131704. 택배상자/택배상자.java
import java.util.*;
import java.io.*;
class Solution {
    /*
    1 2 3 ... n번의 상자가 존재
    왼쪽부터 내림.. 보조 컨테이너 벨트에 보관해야 함..(마지막에 보관한 상자부터 꺼냄)
    
    */
    public int solution(int[] order) {
        Deque<Integer> deque = new ArrayDeque<>();
        int cursor = 0;
        int cnt = 0;
        for(int i = 1; i<=order.length;i++){
            deque.addLast(i);
            while(!deque.isEmpty() && deque.getLast() == order[cursor]){
                deque.removeLast();
                cursor++;
                cnt++;
               
            }
            // System.out.println(order[i]);
        }
        
        return cnt;
    }
}